### DISCUSSION TITLE
🤘🏻 TSX to SFC refactor

### DISCUSSION THREAD

**Discussion by jd-solanki** [jd-solanki/anu#100] [Announcements]
# Why
I am refactoring the whole codebase from TSX to SFC for simplicity and it turns out to be good.

We managed to resolve this #15 issue with this new refactor. ✌🏻

- Simple & clean code, Easy to understand
- Beginner friendly for contribution
- Better IDE & linting support
- Restore the features we lost when we moved to TSX (we weren't using some of the linting rules of vue eslint plugin because of TSX)
- Easier to extract the base component props and extend from it
- much more on the way 😇

Additionally, I am also separating the `ATable` into three variations:
1. `ATable` - Will have simple table markup & styles
3. `ADataTable` - Uses `ATable` as a base. Will perform sorting, filtering, and pagination on passed rows via `rows`. Also supports fetching data from remote.

# ToDos
- [x] Make sure DTS files get generated
- [x] ~Using button with `type="button"` doesn't get styles applied~ (_Invalid_)
- [x] update `types` path in package.json of `anu-vue`
- [x] Restore component meta generation on `nr dev`
- [x] resolve todos
- [x] `ARadio` component's input element also gets the same classes applied to the root input wrapper. Doc example: ListItem slot example
- [x] Close the relevant issue before merging the branch
- [x] Simplify table
  - [x] `ATable`
  - [x] `ADataTable`

**Comment by sherif414** [comment]
Great stuff !!
 I think vue does a lot of compile time optimizations for you when using script setup, keep the good work
